src/sdc11073/xml_types/xml_structure.py

```diff
@@ -914,40 +1043,52 @@ def get_py_value_from_node(self, instance, node):
         except ElementNotFoundException:
             return objects
 
-    def update_xml_value(self, instance, node):
-        """ value is a list of objects with "as_etree_node" method"""
-        # remove all existing nodes
+    def update_xml_value(self, instance: Any, node: etree_.Element):
+        """Write value to node."""
         try:
             py_value = getattr(instance, self._local_var_name)
         except AttributeError:  # set to None (it is in the responsibility of the called method to do the right thing)
             py_value = self._default_py_value
 
         if py_value is not None:
             for val in py_value:
-                #name = self._sub_element_name or val.NODETYPE
                 sub_node = val.as_etree_node(self._sub_element_name, node.nsmap)
                 if hasattr(val, 'NODETYPE') and hasattr(self.value_class, 'NODETYPE') \
                         and val.NODETYPE != self.value_class.NODETYPE:
                     # set xsi type
                     sub_node.set(QN_TYPE, docname_from_qname(val.NODETYPE, node.nsmap))
                 node.append(sub_node)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f'{self.__class__.__name__} datatype {self.value_class.__name__} in subelement {self._sub_element_name}'
 
 
 class ContainerListProperty(_ElementListProperty):
-    """ a list of values that have "mk_node" and cls.from_node methods.
-    Used if maxOccurs="Unbounded" in BICEPS_ParticipantModel"""
+    """ContainerListProperty is a property with a list of elements, each supports xsi:type information.
+
+    Used if maxOccurs="Unbounded" in BICEPS_ParticipantModel.
+    """
 
-    def __init__(self, sub_element_name, value_class, cls_getter, ns_helper, is_optional=True):
+    def __init__(self, sub_element_name: etree_.QName | None,  # Noqa: PLR0913
+                 value_class: Any,
+                 cls_getter: Callable,
```

Comment:
type could be extended to (based on documentation): `Callable[[etree_.QName], typing.Type]`